libsolidity/ast/Types.h

```diff
@@ -991,6 +996,7 @@ class FunctionType: public Type
 	virtual bool isValueType() const override { return true; }
 	virtual bool canLiveOutsideStorage() const override { return m_kind == Kind::Internal || m_kind == Kind::External; }
 	virtual unsigned sizeOnStack() const override;
+	virtual bool hasSimpleZeroValueInMemory() const override { return false; }
```

Comment:
Why cannot the function type be initialised as 0?

Reply:
If you call a zero-initialized internal function pointer, it should assert. If you just set it to zero, the program will jump to zero, which is a valid jumpdest.

Reply:
But how does the zero initialisation code below makes a difference to this?

Reply:
`pushZeroValue` does the magic.

Reply:
Ah I see it inserts functions doing an abort. Nice 😉

Reply:
Hm, do we have a test for this? We should, because it seems to be simple to break this with a careless merge (flipping this bool).

Reply:
I will add one.